src/components/timeline/form/TimelineEditorPanel.svelte

```diff
@@ -978,6 +960,141 @@
+                </div>
+              </RadioButton>
+            </RadioButtons>
+          </Input>
+          <Input layout="inline" class="editor-input">
+            <label for="activity-composition">Labels</label>
+            <RadioButtons
+              selectedButtonId={externalEventOptions.labelVisibility}
+              on:select-radio-button={e => handleExternalEventOptionRadioChange(e, 'labelVisibility')}
+            >
+              <RadioButton use={[[tooltip, { content: 'Always show labels', placement: 'top' }]]} id="on">
+                <div class="radio-button-icon">
+                  <ActivityModeTextIcon />
+                  <span class="timeline-editor-responsive-label">On</span>
+                </div>
+              </RadioButton>
+              <RadioButton use={[[tooltip, { content: 'Never show labels', placement: 'top' }]]} id="off">
+                <div class="radio-button-icon">
+                  <ActivityModeTextNoneIcon />
+                  <span class="timeline-editor-responsive-label">Off</span>
+                </div>
+              </RadioButton>
+              <RadioButton
+                use={[[tooltip, { content: 'Show labels that do not overlap', placement: 'top' }]]}
+                id="auto"
+              >
+                <div class="radio-button-icon">
+                  <ActivityModeWidthIcon />
+                  <span class="timeline-editor-responsive-label">Auto</span>
+                </div>
+              </RadioButton>
+            </RadioButtons>
+          </Input>
+          {#if externalEventOptions.displayMode === 'grouped'}
+            <Input layout="inline" class="editor-input">
+              <label for="activity-composition">Group By</label>
+              <RadioButtons
+                selectedButtonId={externalEventOptions.groupBy}
+                on:select-radio-button={e => handleExternalEventOptionRadioChange(e, 'groupBy')}
+              >
+                <RadioButton
+                  use={[[tooltip, { content: 'Group according to external source', placement: 'top' }]]}
+                  id="source_key"
+                >
+                  <div class="radio-button-icon">
+                    <Truck />
+                    <span class="timeline-editor-responsive-label">By Source</span>
+                  </div>
+                </RadioButton>
+                <RadioButton
+                  use={[[tooltip, { content: 'Group according to event type', placement: 'top' }]]}
+                  id="event_type_name"
+                >
+                  <div class="radio-button-icon">
+                    <Balloon />
+                    <span class="timeline-editor-responsive-label">By Event Type</span>
+                  </div>
+                </RadioButton>
+              </RadioButtons>
+            </Input>
+            <Input layout="inline" class="editor-input">
+              <label for="activity-composition">Bin Size</label>
```

Comment:
Curious - what is the use case for exposing a non-default bin size input?

Reply:
I think our idea is that since it's totally normal to have _a lot_ of external events (if you're planning over 7+ days, hundreds reaching towards thousands is absolutely expected), bin size helps create "dumb" organization on-top of our other grouping options ("dumb" as in, it doesn't group on any specific filter, just every X).

Reply:
Sounds pretty similar to activities which can also range anywhere from a few to hundreds of thousands (including activities that decompose into thousands of instances). What is the use case for allowing the user to tweak the number of bins? The binning is recursive so if there are actually thousands of items to be binned then sub-bins will be made. Pretty much the same thing as what Chrome console does with very large arrays.

Reply:
Our reasoning was just that events could have a lot of occurrences, so a change from the default of 100 that exists for activities might make sense as we could be operating on a different scale with imported activities. In any case, we have removed the feature as of [this commit](https://github.com/NASA-AMMOS/aerie-ui/pull/1396/commits), unless we should keep it @duranb  ?